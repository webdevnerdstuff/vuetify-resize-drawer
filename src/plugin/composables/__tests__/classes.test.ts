import { describe, it, expect } from 'vitest';
import {
	useDrawerClasses,
	useHandleContainerClasses,
	useHandleIconClasses,
} from '../classes';


describe('Classes Composable', () => {
	describe('useDrawerClasses', () => {
		it('should return class object', () => {
			const classes = useDrawerClasses({
				absolute: false,
				expandOnHover: false,
				floating: null,
				isMouseover: false,
				location: 'start',
				rail: false,
				railWidth: '8',
				temporary: undefined,
			});

			expect(classes).toMatchSnapshot();
		});
	});

	describe('useHandleContainerClasses', () => {
		it('should return class object', () => {
			const classes = useHandleContainerClasses({
				drawerLocation: 'start',
				handlePosition: 'center',
			});

			expect(classes).toMatchSnapshot();
		});

		it('should return class object', () => {
			const classes = useHandleContainerClasses({
				drawerLocation: 'left',
				handlePosition: 'top',
			});

			expect(classes).toMatchSnapshot();
		});
	});

	describe('useHandleIconClasses', () => {
		it('should return class object with fa classes as false', () => {
			const classes = useHandleIconClasses({
				drawerLocation: 'start',
				handlePosition: 'top',
				iconOptions: {
					defaultSet: 'mdi'
				},
				isUserIcon: false,
			});

			expect(classes).toMatchSnapshot();
		});

		it('should return class object with fa classes as true', () => {
			const classes = useHandleIconClasses({
				drawerLocation: 'left',
				handlePosition: 'center',
				iconOptions: {
					defaultSet: 'fa'
				},
				isUserIcon: false,
			});

			expect(classes).toMatchSnapshot();
		});
	});
});
