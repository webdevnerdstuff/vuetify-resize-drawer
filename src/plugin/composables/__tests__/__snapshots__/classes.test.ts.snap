// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Classes Composable > useDrawerClasses > should return class object 1`] = `
{
  "v-navigation-drawer--absolute": false,
  "v-navigation-drawer--custom-rail": true,
  "v-navigation-drawer--fixed": true,
  "v-navigation-drawer--floating": null,
  "v-navigation-drawer--is-mouseover": false,
  "v-navigation-drawer--left": true,
  "v-navigation-drawer--open-on-hover": false,
  "v-navigation-drawer--rail": false,
  "v-navigation-drawer--right": false,
  "v-navigation-drawer--temporary": false,
  "v-resize-drawer": true,
  "v-resize-drawer--start": false,
}
`;

exports[`Classes Composable > useHandleContainerClasses > should return class object 1`] = `
{
  "v-navigation-drawer__handle": true,
  "v-resize-drawer--handle-container": true,
  "v-resize-drawer--handle-container-parent-start": true,
  "v-resize-drawer--handle-container-position-center": true,
}
`;

exports[`Classes Composable > useHandleContainerClasses > should return class object 2`] = `
{
  "v-navigation-drawer__handle": true,
  "v-resize-drawer--handle-container": true,
  "v-resize-drawer--handle-container-parent-left": true,
  "v-resize-drawer--handle-container-position-top": true,
}
`;

exports[`Classes Composable > useHandleIconClasses > should return class object with fa classes as false 1`] = `
{
  "v-resize-drawer--handle-container-icon": true,
  "v-resize-drawer--handle-container-icon-fa": false,
  "v-resize-drawer--handle-container-icon-fa-top": false,
  "v-resize-drawer--handle-container-icon-top-start": true,
  "v-resize-drawer--handle-container-icon-user-icon": false,
}
`;

exports[`Classes Composable > useHandleIconClasses > should return class object with fa classes as true 1`] = `
{
  "v-resize-drawer--handle-container-icon": true,
  "v-resize-drawer--handle-container-icon-center-left": true,
  "v-resize-drawer--handle-container-icon-fa": true,
  "v-resize-drawer--handle-container-icon-fa-center": true,
  "v-resize-drawer--handle-container-icon-user-icon": false,
}
`;
